rfcs/0145-doc-strings.md

```diff
@@ -0,0 +1,429 @@
+---
+feature: doc-comment-standard
+start-date: 2023-03-27
+author: hsjobeki
+co-authors: --
+shepherd-team: @DavHau; @sternenseemann; @asymmetric
+shepherd-leader: @lassulus
+related-issues: (will contain links to implementation PRs)
+---
+
+# Summary
+[summary]: #summary
+
+Propose a standard format specification for doc-comments.
+
+This RFC includes two concerns that define a doc-comment:
+
+- Outer format rules to allow distinction between regular comments and doc-comments
+- Inner format rules that describe the required format of a doc-comment.
+
+However, both concerns relate closely to each other; It makes sense and reduces bureaucracy to address both in a single RFC.
+
+# Definitions
+
+For this RFC, we adopt the following definitions:
+
+- **doc-comment**: A structured comment documenting the code's API.
+- **internal comment**: A free-form comment for those interested in the code's implementation.
+
+The doc-comment properties are grouped into these subcategories:
+
+- **Outer format**: Specifies rules linking code (API) and doc-comments. (e.g. placement, syntax rules)
+
+- **Inner format**: Specifies rules affecting the comment's actual content. (e.g. usage of commonMark)
+
+# Motivation
+[motivation]: #motivation
+
+The following are the envisioned goals.
+
+- Create distinct outer and inner formats for Nix doc-comments to enable accurate automated parsing and extraction. 
+- Ensure clear differentiation from internal comments, making them accessible to tooling solutions such as documentation rendering and building. 
+
+- Migrate `nixpkgs'` comments to this format.
```

Comment:
```suggestion
- Migrate `nixpkgs` comments (when they were meant as doc-comments) to this format.
```

Reply:
I did open this (draft) PR. https://github.com/NixOS/nixpkgs/pull/262987 which showcases the total migration effort.
Its hard to detect (with tooling) if something is meant to be a doc-comment. But i think we can just do some manual work. Starting with the lib functions.